cmd/api/main.go

```diff
@@ -169,6 +171,7 @@ func main() {
 			grpc_zap.UnaryServerInterceptor(grpcLogger, zapOpts...),
 			grpc_auth.UnaryServerInterceptor(determineAuth),
 			prometheus.UnaryServerInterceptor,
+			fieldmask.UnaryServerInterceptor(),
```

Comment:
shouldn't we have a feature enablement check of some sort ?  presumably a client could send the extra headers or encoded data and it is OK to ignore ?

Reply:
I have added a feature flag implementation to handle all features control.

Reply:
feature gate looks good @sayan-biswas good think .... in looking at it though I raised some questions around changes you made to my recently added tuning options 

I'll submit a separate review for those